lua_modules/http/httpserver.lua

```diff
@@ -116,13 +116,12 @@ do
       local ondata = function(conn, chunk)
         -- feed request data to request handler
         if not req or not req.ondata then return end
-        req:ondata(chunk)
-        -- NB: once length of seen chunks equals Content-Length:
-        --   onend(conn) is called
         body_len = body_len + #chunk
-        -- print("-B", #chunk, body_len, cnt_len, node.heap())
         if body_len >= cnt_len then
-          req:ondata()
+          -- last chunk
+          req:ondata(chunk, true)
+        else
+          req:ondata(chunk, false)
```

Comment:
What's wrong with calling `req:ondata()` with a `nil` `chunk` to indicate the end of the data?  (I realize that `req:ondata()` is perhaps confusing, but it is equivalent to `req:ondata(nil)`.)

Reply:
Call once  maybe batter then call twice when the last chunk was came

Reply:
Oh, that's an interesting counter-point... hm.  I *think* I still like `nil` better, but would appreciate hearing opinions of more proficient Lua programmers (@TerryE or @pjsg or @djphoenix spring to mind).

Reply:
I would still like a 2nd opinion on the API here.